### DISCUSSION TITLE
The finally block by itself without having try or catch

### DISCUSSION THREAD

**Discussion by TizzyT566** [dotnet/csharplang#5020] [Language Ideas] [answered]
There is a method of using the try{}finally{} to prevent cases where Thread.Abort() is called, having the code within finally not be interrupted. I was wondering if this should instead be written as finally{} omitting the try{}. Use case may be so minimal that its probably not granted but just a thought.

**Comment by Joe4evr** [comment, marked as answer]
`Thread.Abort()` is no longer supported and will throw PNSE on all platforms, so your code should not have to worry about being aborted.

**Reply by yaakov-h** [reply]
Only in .NET >= 5 I believe... though if you're code is new enough to use a new language feature being proposed here, it will be new enough for Thread.Abort to not be a problem 😅

**Reply by 333fred** [reply]
No. What's new in 5 is that you get a compile-time warning for it. The API has never worked on core.

**Reply by Joe4evr** [reply]
And even if you were running on a "full framework" version where it did function, from what I understand the use of `Thread.Abort()` is almost always certainly the wrong thing anyway.
